app/pulp/app/models/task.py

```diff
@@ -35,6 +35,30 @@ class ReservedResource(Model):
     worker = models.ForeignKey("Worker", on_delete=models.CASCADE, related_name="reservations")
 
 
+class WorkerManager(models.Manager):
+    def get_unreserved_worker(self):
+        """
+        Find an unreserved Worker
```

Comment:
Something like this for example:  https://github.com/bmbouter/pulp/blob/e806ab8a11a10fd1bbd9c7475f3187f0c8157d43/app/pulp/app/serializers/base.py#L86-L86